### PR TITLE
feat: added script to transform postman collections to OpenApiV3

### PR DESCRIPTION
**Pull request by bahrmichael**
This adds a script that takes Postman Collections, and turns them into OpenAPI V3 yamls. No strong opinion on the name of the script, happy to incorporate your feedback! :)

I will soon follow with a PR where I turn the generated OpenAPI into a json schema.

### CODE REVIEW

**Comment by bahrmichael** [comment]
Looks like there's an issue with the flexport validation @RafalWilinski. I should be on the latest main commit.

File: src/postman-2-openapi.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    const doc = yaml.load(schema);
+
+    await asyncRimraf('./generated');
+    mkdirSync(`./generated`, { recursive: true });
+    const outputFile = `./generated/${provider}.json`;
+    writeFileSync(outputFile, JSON.stringify(doc, null, 2));
+}
+
+(async () => {
+    await generate("./postman_collections", "quickbooks", "postman_collection.2022-03-23.json");
```
Comment (RafalWilinski) [inline src/postman-2-openapi.ts:25]: I'm unsure about keeping the actual postman collection in the repo - can we point to a source instead?

Comment (bahrmichael) [reply src/postman-2-openapi.ts:25]: Currently I wouldn't know how, but I'll research a bit more. On their website there's a postman data string that I can investigate a bit more. Best case would be that we scrape their website, extract the postman data string and somehow convert that to a postman collection (currently done through the client).

**Comment by bahrmichael** [comment]
Tried to use the Postman API to download the collection via the ID found in the attribute `data-postman-var-1` in their Postman Run button, but am receiving

```
{
    "error": {
        "name": "instanceNotFoundError",
        "message": "We could not find the collection you are looking for"
    }
}
```

The other interesting attribute `data-postman-param` does not hold information that looks like an ID:

```
[{"type":"text","enabled":true,"key":"baseurl","value":"sandbox-quickbooks.api.intuit.com"},{"type":"text","enabled":true,"key":"companyid","value":"<<companyId/realmId>>"},{"type":"text","enabled":true,"key":"minorversion","value":"14"},{"type":"text","enabled":true,"key":"UserAgent","value":"QBOV3-OAuth2-Postman-Collection"}]
```

**Comment by bahrmichael** [comment]
When importing a collection via the Run in Postman button I don't see any logs.

**Comment by bahrmichael** [comment]
The API I use (you need a postman api key), is `https://api.getpostman.com/collections/c4c9e1618576a68ac526`. I can see that the run button tells the postman app to use that collection as well.

**Comment by bahrmichael** [comment]
Same when running with the `newman` cli:

```
newman run "https://api.getpostman.com/collections/c4c9e1618576a68ac526?apikey=PMAK-redacted" --export-collection collection.json
error: collection could not be loaded
  Error fetching the collection from the provided URL. Ensure that the URL is valid.
  We could not find the collection you are looking for
```

**Comment by bahrmichael** [comment]
Same problem with `postmanctl`:

```
./postmanctl fork collection c4c9e1618576a68ac526 -l quickbooks
error: status code: 404, name: instanceNotFoundError, message: The specified item does not exist., details map[id:c4c9e1618576a68ac526 item:collection]
```

Here I noticed however that if I forked a collection once, I can now access it from my workspace. So if we're okay with manually updating the collection in our own postman workspace, we could export it from there.

```
./postmanctl get collections
UID                                             NAME
20203465-0d873e1c-9d0b-4ca0-ac53-33a26db8180f   Team Transfer
20203465-a1bfe895-e3b1-4549-aa89-345ff26285a7   QuickBooks Online V3API Collections - OAuth2
20203465-fcbf820c-8f47-498c-bef4-54d81c6e2c06   New Collection
```

**Comment by bahrmichael** [comment]
Good news! Exporting the forked collection works:

```
newman run "https://api.getpostman.com/collections/redacted?apikey=PMAK-redacted" --export-collection collection.json
```

```
head collection.json
{
  "_": {
    "postman_id": "redacted"
  },
  "item": [
    {
      "id": "redacted",
      "name": "Account",
      "description": {
        "content": "An Account object is a component of a Chart Of Accounts, and is part of a ledger. It is used to record a total monetary amount allocated against a specific use. Accounts are one of five basic types: asset, liability, revenue (income), expenses, or equity. Delete is achieved by setting the Active attribute to false in an object update request; thus, making it inactive. In this type of delete, the record is not permanently deleted, but is hidden for display purposes. References to inactive objects are left intact. \n\nhttps://developer.intuit.com/docs/api/accounting/account",
```

**Comment by bahrmichael** [comment]
It looks like `newman` strips the raw url, which `postman-to-openapi` requires. Fetching with `axios` doesn't cause that problem.

File: src/postman-2-openapi.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    const outputFile: string = `./generated/${provider}.json`;
+    writeFileSync(outputFile, JSON.stringify(jsonSchema, null, 2));
+}
+
+(async () => {
+    if (!POSTMAN_API_KEY) {
+        throw Error(`Environment variable POSTMAN_API_KEY is missing.`);
+    }
+
+    await generate("20203465-a1bfe895-e3b1-4549-aa89-345ff26285a7", "quickbooks", POSTMAN_API_KEY);
```
Comment (bahrmichael) [inline src/postman-2-openapi.ts:43]: This collection currently lives in my personal postman workspace. Let's discuss if we want to fork the quickbooks collection it into a stedi workspace.

File: src/postman-2-openapi.ts (hunk 1 above)
Comment (bahrmichael) [reply src/postman-2-openapi.ts]: Solved, see below.

**Review by RafalWilinski — APPROVED** [review]
Lets merge this and figure out small things like where the workspace should live later. I think it's in good shape to start experimenting

**Pull request merged**